src/pages/gateway/fastly.md

```diff
@@ -0,0 +1,257 @@
+     - **Name** - api_mesh_miss
+     - **Type** - **miss**
+     - **Priority** - **60**
+     - **Content**:
+
+        ```csharp
+        if (req.url~"^/api/") {
+          set req.backend = F_graph_prod_adobe_io;
+        }
+        //API Mesh prod mapping
+        if (req.url~"^/api/<mesh_id>") {
+          set bereq.http.x - api - key = "<mesh_api_key>";
+        }
+        # //Optionally add another environment
+        # //API Mesh stage mapping
+        # if (req.url~"^/api/<mesh_id>") {
+        #   set bereq.http.x - api - key = "<mesh_api_key>";
+        # }
+        ```
+
+  The following `api_mesh_pass` snippet allows you to query your mesh URL without appending the `api_key`:
+
+   - Cache pass - replace the `<mesh_id>` and `<mesh_api_key>` placeholders with the information from your mesh URL, which has the following structure: `https://graph.adobe.io/api/<meshId>/graphql?api_key=<mesh_api_key>`:
+     - **Name** - api_mesh_pass
+     - **Type** - **pass**
+     - **Priority** - **10**
+     - **Content**:
+
+        ```csharp
+        if (req.url ~ "^/api/") {
+          set req.backend = F_graph_prod_adobe_io;
+        }
+        //API Mesh prod mapping
+        if (req.url ~ "^/api/<mesh_id>") {
+          set bereq.http.x-api-key = "<mesh_api_key>";
+        }
+        # //Optionally add another environment
+        # //API Mesh stage mapping
+        # if (req.url ~ "^/api/<mesh_id>") {
+        #   set bereq.http.x-api-key = "<mesh_api_key>";
+        # }
+        ```
+
+   - Cache fetch:
+     - **Name** - api_mesh_fetch
+     - **Type** - **fetch**
+     - **Priority** - **10**
+     - **Content**:
+
+        ```csharp
+        if (req.http.x - commerce - bypass - fastly - cache == "true") {
+          return (pass);
+        }
+        ```
+
+   - Cache deliver:
+     - **Name** - api_mesh_deliver
+     - **Type** - **deliver**
+     - **Priority** - **10**
+     - **Content**:
+
+        ```csharp
+        if (req.http.x - commerce - bypass - fastly - cache == "true") {
+          return (deliver);
+        }
+        ```
+
+1. In **Fastly Configuration** > **Backend Settings** click **Create**. Add a new backend with the following information:
+
+   - **Condition** - `req.url ~ "^/api/(.*)" ,`
+   - **Address** - **graph.adobe.io**
+   - **Priority** - **8**
+
+1. In **Fastly Configuration** click **Upload VCL to Fastly**. Click **Save Config**.
+
+## Configure headers in API Mesh
+
+<InlineAlert variant="info" slots="text"/>
+
+After adding VCL snippets in the [Fastly Setup](#configure-fastly-in-adobe-commerce), your Commerce GraphQL URL is now in the following format with no `api_key` appended: `https://graph.adobe.io/api/<meshId>/graphql`
+
+To distinguish between requests from users and requests from API Mesh, use the following source operation header to prevent Fastly from caching headers that come directly from API Mesh:
+
+```json
+"x-commerce-bypass-fastly-cache": "true" 
```

Comment:
we can explain here that this prevents the double caching problem
it's worth including a diagram..
that helped immediately to anybody that I showed this to - they immediately understood it